src/app/components/media/MediaTimeline.js

```diff
@@ -0,0 +1,469 @@
+        }
+      }
+
+      entities[`place-${name}`].instances.push({ id, start_seconds, end_seconds });
+    });
+
+    const data = {
+      project: {
+        projectclips: projecttags,
+        projecttags,
+        projectplaces: projecttags,
+      },
+      commentThreads,
+      videoClips: Object.values(entities).filter(({ project_clip }) => !!project_clip),
+      videoTags: Object.values(entities).filter(({ project_tag }) => !!project_tag),
+      videoPlaces: Object.values(entities).filter(({ project_place }) => !!project_place),
+      user: {
+        first_name,
+        id: userId,
+        last_name: '',
+        profile_img_url,
+      },
+    };
+
+    console.log({ data });
+
+    return {
+      duration, time, data, mediaId, dbid,
+    };
+  }
+
+  componentDidUpdate() {
+    const { fragment: { id } = {} } = this.props;
+    const instance = id ? document.querySelector(`*[data-instance-id="clip-${id.trim()}"]`) : null;
+
+    const outline = '2px solid #2e77fc';
+
+    if (instance && instance.style.outline !== outline) {
+      instance.style.outline = outline;
+      instance.scrollIntoView();
+    }
+  }
+
+  commentThreadCreate = (time, text, callback) => {
+    const { mediaId, dbid } = this.state;
+    createCommentThread(text, `t=${time}`, `${dbid}`, mediaId, callback);
+  };
+
+  commentThreadDelete = (id, callback) => {
+    const { dbid } = this.state;
+    destroyComment(id, dbid, callback);
+  };
+
+  commentCreate = (threadId, text, callback) => {
+    const { data: { commentThreads } } = this.state;
+    const id = commentThreads.find(({ id }) => id === threadId).dbid;
+    createComment(text, null, `${id}`, threadId, callback);
+  };
+
+  commentEdit = (threadId, commentId, text, callback) => updateComment(commentId, text, callback);
+
+  commentDelete = (threadId, commentId) => destroyComment(commentId, threadId);
+
+  instanceClip = (type, entityId, instanceId) => {
+    const { mediaId, dbid, data: { videoTags, videoPlaces, videoClips } } = this.state;
+
+    const entity = (type === 'tag' ? videoTags : videoPlaces).find(({ id }) => id === entityId);
+    const { name, instances } = entity;
+
+    const { start_seconds, end_seconds } = instances.find(({ id }) => id === instanceId);
+    console.log({ start_seconds, end_seconds });
+
+    const clipEntity = videoClips.find(c => c.name === name);
+    const clips = clipEntity ? clipEntity.instances : [];
+    console.log({ clips });
+
+    const intervals = [...clips.map(({ id, start_seconds, end_seconds }) =>
+      [start_seconds, end_seconds, id]), [start_seconds, end_seconds, null]];
+    console.log(JSON.stringify(intervals));
+
+    const segments = intervals.sort((a, b) => a[0] - b[0])
+      .reduce((ac, x) => (!ac.length || ac[ac.length - 1][1] < x[0]
+        ? ac.push(x)
+        : ac[ac.length - 1][1] = Math.max(ac[ac.length - 1][1], x[1]), ac), []);
+
+    console.log({ segments });
```

Comment:
Please remove it when you don't need it anymore.

Reply:
fixed